coremltools/models/deprecation.py

```diff
@@ -0,0 +1,23 @@
+import warnings
+import functools
+
+def deprecated(func=None, str=""):
+    '''
+    Decorator to mark a function or class as deprecated
```

Comment:
Does it really work for classes? It looks like you're doing something else for classes.

Reply:
Right now its only for functions, updated the comment.

Reply:
I think it would be good to have one decorator that works for functions, classes or any other object type. 

In fact, I think this decorator should work for classes as it is.

Reply:
Yes it does